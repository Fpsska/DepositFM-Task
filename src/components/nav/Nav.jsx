import React from 'react';
import { useSelector } from 'react-redux';

import NavLink from './NavLink';
import './nav.scss';


const Nav = () => {
    const { navLinks } = useSelector(state => state.navReducer);
    // 
    return (
        <nav className="nav">
            <ul className="nav__menu">
                {navLinks.map(item => {
                    return (
                        <NavLink
                            key={item.id}
                            id={item.id}
                            text={item.text}
                            link={item.link}
                            isActive={item.isActive}
                        />
                    );
                })}
            </ul>
        </nav>
    );
};

export default Nav;