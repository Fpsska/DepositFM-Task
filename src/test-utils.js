import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { BrowserRouter } from 'react-router-dom';

import navSlice from './app/slices/navSlice';
import formSlice from './app/slices/formSlice';
import paletteSlice from './app/slices/paletteSlice';

// /. imports

const render = (
    ui,
    { preloadedState, store = configureStore({ reducer: { navSlice, formSlice, paletteSlice }, preloadedState }),
        ...renderOptions } = {}
) => {

    function Wrapper({ children }) {
        return (
            <BrowserRouter>
                <Provider store={store}>{children}</Provider>
            </BrowserRouter>
        );
    }

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };