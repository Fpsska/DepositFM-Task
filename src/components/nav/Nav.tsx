import React from 'react';

import { useAppSelector } from '../../app/hooks';

import NavLink from './NavLink';

import './nav.scss';

// /. imports

const Nav: React.FC = () => {
    const { navLinks } = useAppSelector(state => state.navSlice);
    const { isPreloaderVisible } = useAppSelector(state => state.formSlice);

    return (
        <nav className="nav">
            <ul className="nav__menu">
                {navLinks.map(item => {
                    return (
                        <NavLink
                            key={item.id}
                            {...item}

                            isPreloaderVisible={isPreloaderVisible}
                        />
                    );
                })}
            </ul>
        </nav>
    );
};

export default Nav;