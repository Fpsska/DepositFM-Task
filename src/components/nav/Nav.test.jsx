import { render, screen, fireEvent } from '../../test-utils';

import Nav from './Nav';

// /. imports

const NavData = [
    {
        id: 1,
        text: 'Form',
        link: '/DepositFM-Task/',
        isActive: true
    }
];

describe('Nav component', () => {
    test('render component snapshot with data', () => {
        const { container } = render(<Nav data={NavData} />);

        expect(container).toMatchSnapshot();
    });
    test('render component empty snapshot', () => {
        const { container } = render(<Nav />);

        expect(container).toMatchSnapshot();
    });
    test('should be set InnerHTML, href for element', () => {
        render(<Nav data={NavData} />);

        expect(screen.getAllByTestId('nav-link')[0]).toHaveTextContent(NavData[0].text);
        expect(screen.getAllByTestId('nav-link')[0]).toHaveAttribute('href', NavData[0].link);
    });
    test('should be set active class for first link', () => {
        render(<Nav data={NavData} />);

        expect(screen.getAllByTestId('nav-link')[0]).toHaveClass('nav__link active');
    });
    test('should be set active class for current clicked link element', () => {
        render(<Nav data={NavData} />);

        // fireEvent.click(screen.getAllByTestId('nav-link')[1]);
        // expect(screen.getAllByTestId('nav-link')[1]).toHaveClass('nav__link active');
        // expect(screen.getAllByTestId('nav-link')[0]).toHaveClass('nav__link');
    });
});

