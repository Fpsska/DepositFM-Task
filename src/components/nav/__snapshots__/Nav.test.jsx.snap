// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component render component empty snapshot 1`] = `
<div>
  <nav
    class="nav"
  >
    <ul
      class="nav__menu"
    >
      <li
        class="nav__item"
      >
        <a
          class="nav__link active"
          data-testid="nav-link"
          href="/DepositFM-Task/"
        >
          Form
        </a>
      </li>
      <li
        class="nav__item"
      >
        <a
          class="nav__link"
          data-testid="nav-link"
          href="/Palette"
        >
          Palette
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Nav component render component snapshot with data 1`] = `
<div>
  <nav
    class="nav"
  >
    <ul
      class="nav__menu"
    >
      <li
        class="nav__item"
      >
        <a
          class="nav__link active"
          data-testid="nav-link"
          href="/DepositFM-Task/"
        >
          Form
        </a>
      </li>
      <li
        class="nav__item"
      >
        <a
          class="nav__link"
          data-testid="nav-link"
          href="/Palette"
        >
          Palette
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
